src/lib/features/playground/playground-service.ts

```diff
@@ -68,39 +92,66 @@ export class PlaygroundService {
                     ? new Date(context.currentTime)
                     : undefined,
             };
-            const output: PlaygroundFeatureSchema[] = await Promise.all(
```

Comment:
extracting into a shared method that will be reused in advanced query evaluation